xwiki-platform-core/xwiki-platform-user/xwiki-platform-user-profile/xwiki-platform-user-profile-ui/src/main/resources/XWiki/XWikiUserPreferencesSheet.xml

```diff
@@ -37,6 +37,164 @@
+        <disabled>0</disabled>
+        <displayFormType>select</displayFormType>
+        <displayType>yesno</displayType>
+        <name>parse</name>
+        <number>4</number>
+        <prettyName>Parse content</prettyName>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.BooleanClass</classType>
+      </parse>
+      <use>
+        <cache>0</cache>
+        <disabled>0</disabled>
+        <displayType>select</displayType>
+        <freeText>forbidden</freeText>
+        <largeStorage>0</largeStorage>
+        <multiSelect>0</multiSelect>
+        <name>use</name>
+        <number>3</number>
+        <prettyName>Use this extension</prettyName>
+        <relationalStorage>0</relationalStorage>
+        <separator> </separator>
+        <separators>|, </separators>
+        <size>1</size>
+        <unmodifiable>0</unmodifiable>
+        <values>currentPage|onDemand|always</values>
+        <classType>com.xpn.xwiki.objects.classes.StaticListClass</classType>
+      </use>
+    </class>
+    <property>
+      <cache>forbid</cache>
+    </property>
+    <property>
+      <code>require(['jquery', 'xwiki-meta'], function ($, xm) {
+  /**
+   * Handle the shortcut preference modifications
+   */
+  $(document).off('input.shortcutPreference');
+  let updateShortcut = function (shortcutPreferenceDOM) {
+    let shortcutPreference = $(shortcutPreferenceDOM);
+    let translateKey = shortcutPreference.attr("data-translatekey");
+    let newValue = shortcutPreference.val();
+    let url = xm.restURL;
+    let oldItemNumber = shortcutPreference.attr("data-shortcutprefnumber");
+    if (oldItemNumber == null) {
+      if (newValue !== shortcutPreference.attr("data-defaultvalue")) {
+        // Create the new user preference
+        let creationUrl = url + "/objects";
+        let data = {
+          'className': 'XWiki.UserPreferenceShortcutsClass',
+          'property#translationKey': translateKey,
+          'property#value': newValue
+        };
+        $.post(creationUrl, data, function(newObject) {
+          shortcutPreference.attr("data-shortcutprefnumber", newObject.firstChild.getElementsByTagName("number")[0].textContent);
+        });
+      }
+    } else {
+      if (newValue == shortcutPreference.attr("data-defaultvalue")) {
+        // Remove the existing preference, we don't want to keep it if the user is back to normal.
+        let destructionUrl = url + "/objects/XWiki.UserPreferenceShortcutsClass/" + oldItemNumber;
+        $.ajax({url: destructionUrl,
+          type: "DELETE"
+        });
+        shortcutPreference.removeAttr("data-shortcutprefnumber");
+      } else {
+        // Update the existing preference
+        let updateUrl = url + "/objects/XWiki.UserPreferenceShortcutsClass/" + oldItemNumber +"/properties/value";
+        let data = {
+          'property#value': newValue
+        };
+        $.ajax({url: updateUrl,
+          type: "PUT",
+          data: data
+        });
```

Comment:
It might be good to handle errors here and maybe also display a notification if the changes have been saved. Otherwise, users have no way to understand that there was an error saving the keyboard shortcuts. See, e.g., https://github.com/xwiki/xwiki-platform/blob/05f0ba0694a3e5e4395f493c3388ac152d218ad4/xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-war/src/main/webapp/resources/uicomponents/edit/editableProperty.js#L156-L178 for an example how this is handled in other places.